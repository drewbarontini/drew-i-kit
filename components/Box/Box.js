import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { ifProp } from 'styled-tools';

import { setProp, palette, theme } from '../../lib/styleUtils';

const Box = styled.div`
  ${setProp({
    prop: 'backgroundColor',
    cssProp: 'background-color',
    themeKey: 'colors',
  })};

  ${props =>
    ifProp(
      'border',
      css`
        border: ${theme('borders.sizes.base')(props)} solid
          ${palette(`${props.border}`)(props)};
      `
    )}

  ${setProp({ prop: 'bottom', themeKey: 'spacing' })};
  ${setProp({ prop: 'height' })};
  ${setProp({ prop: 'left', themeKey: 'spacing' })};
  ${setProp({ prop: 'padding', themeKey: 'spacing' })};
  ${setProp({ prop: 'position' })};
  ${setProp({ prop: 'right', themeKey: 'spacing' })};
  ${setProp({ prop: 'top', themeKey: 'spacing' })};
  ${setProp({ prop: 'width' })};
`;

Box.propTypes = {
  backgroundColor: PropTypes.string,
  bottom: PropTypes.string,
  height: PropTypes.string,
  left: PropTypes.string,
  padding: PropTypes.string,
  position: PropTypes.string,
  right: PropTypes.string,
  top: PropTypes.string,
  width: PropTypes.string,
};

Box.displayName = 'Box';

export default Box;
