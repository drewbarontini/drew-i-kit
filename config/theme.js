const theme = {
  borders: {
    radius: {
      base: '3px',
    },
  },
  colors: {
    bg: '#fff',
    fg: '#272727',
    primary: '#3576a7',
    subdued: '#cbcbcb',
    error: '#fe5f55',
    success: '#8acb88',
    warning: '#ffc527',
  },
  fonts: {
    family: {
      base:
        "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
      heading:
        "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
    },
    lineHeight: {
      base: '1.5',
      heading: '1.2',
    },
    sizes: {
      base: '14px',
      xxs: '8px',
      xs: '10px',
      s: '12px',
      l: '16px',
      xl: '24px',
      h1: '26px',
    },
    weights: {
      light: 200,
      normal: 400,
      semiBold: 600,
      bold: 800,
    },
  },
  spacing: {
    base: '16px',
    xs: '5px',
    s: '10px',
    m: '12px',
    l: '32px',
    xl: '60px',
  },
};

export default theme;
