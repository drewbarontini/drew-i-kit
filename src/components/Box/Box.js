import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { ifProp } from 'styled-tools';

import { setProp, palette, theme } from '../../lib/props';
import { responsive } from '../../lib/responsive';

const Box = styled.div`
  ${setProp({
    prop: 'backgroundColor',
    themeKey: 'colors',
  })};

  ${props =>
    ifProp(
      'border',
      css`
        border: ${theme('borders.sizes.base')(props)} solid
          ${palette(`${props.border}`)(props)};
      `
    )}

  ${setProp({ prop: 'bottom', themeKey: 'spacing' })};
  ${setProp({ prop: 'display' })}
  ${setProp({ prop: 'height' })};
  ${setProp({ prop: 'left', themeKey: 'spacing' })};
  ${setProp({ prop: 'padding', themeKey: 'spacing' })};
  ${setProp({ prop: 'position' })};
  ${setProp({ prop: 'right', themeKey: 'spacing' })};
  ${setProp({ prop: 'top', themeKey: 'spacing' })};
  ${setProp({ prop: 'width' })};

  ${responsive};
`;

Box.propTypes = {
  backgroundColor: PropTypes.string,
  bottom: PropTypes.string,
  display: PropTypes.string,
  height: PropTypes.string,
  left: PropTypes.string,
  padding: PropTypes.string,
  position: PropTypes.string,
  right: PropTypes.string,
  top: PropTypes.string,
  width: PropTypes.string,
};

Box.displayName = 'Box';

export default Box;
