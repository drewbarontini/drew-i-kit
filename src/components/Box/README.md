# Box

### Usage

```jsx
import { Box } from 'drew-i-kit';

<Box height="200px" width="200px" backgroundColor="primary" />
```

```jsx
import { Box } from 'drew-i-kit';

<Box height="200px" width="200px" backgroundColor="primary" />
<Box padding="base">
  <p>Some content here...</p>
</Box>
<Box padding="base" textAlign="center">
  <p>Some content here...</p>
</Box>
```

### Props

- `backgroundColor` (**string**, optional)
- `border` (**Boolean**, optional)
- `bottom` (**string**, optional)
- `display` (**string**, optional)
- `height` (**string**, optional)
- `left` (**string**, optional)
- `padding` (**string**, optional)
- `position` (**string**, optional)
- `right` (**string**, optional)
- `textAlign` (**string**, optional)
- `top` (**string**, optional)
- `width` (**string**, optional)