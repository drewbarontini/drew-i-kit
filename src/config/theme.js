import colors from './colors';
import fonts from './fonts';
import icons from './icons';

const theme = {
  borders: {
    sizes: {
      base: '1px',
      l: '2px',
    },
    radius: {
      base: '3px',
    },
  },
  breakpoints: {
    sizes: {
      s: 380,
      m: 768,
      l: 900,
      xl: 1100,
    },
    query: 'max-width',
    unit: 'px',
  },
  colors: {
    ...colors,
  },
  fonts: {
    ...fonts,
  },
  icons: {
    ...icons,
  },
  shadows: {
    base: '0 3px 3px rgba(0, 0, 0, 0.25);',
  },
  sizing: {
    base: '900px',
    s: '400px',
    l: '1200px',
  },
  spacing: {
    base: '20px',
    xs: '5px',
    s: '10px',
    l: '40px',
    xl: '60px',
  },
  transition: {
    base: '0.3s ease-in-out',
  },
};

export default theme;
