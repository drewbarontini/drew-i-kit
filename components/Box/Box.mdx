---
name: Box
route: /components/box
---

# Box

import { Playground } from 'docz';
import Box from '../Box';

<Playground>
  <Box height="200px" width="200px" backgroundColor="primary" />
</Playground>
