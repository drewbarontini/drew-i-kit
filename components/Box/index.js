import withThemeProvider from '../withThemeProvider';
import Box from './Box';

export default withThemeProvider(Box);
