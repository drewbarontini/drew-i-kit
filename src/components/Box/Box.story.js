import React from 'react';
import { storiesOf } from '@storybook/react';

import Box from '.';

storiesOf('Box', module).add('default', () => (
  <Box width="200px" height="200px" backgroundColor="primary.base" />
));
