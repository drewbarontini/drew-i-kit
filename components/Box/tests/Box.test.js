import React from 'react';
import { render } from 'react-testing-library';
import 'jest-styled-components';

import theme from '../../../config/theme';

import Box from '../';

describe('<Box />', () => {
  it('should render properly', () => {
    const { container } = render(
      <Box backgroundColor="primary" padding="base" theme={theme} />
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule(
      'background-color',
      theme.colors.primary
    );
    expect(container.firstChild).toHaveStyleRule('padding', theme.spacing.base);
  });
});
