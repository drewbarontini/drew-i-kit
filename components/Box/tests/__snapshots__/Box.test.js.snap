// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> should match snapshot 1`] = `
.c0 {
  background-color: #3576a7;
  padding: 20px;
}

<div
  class="c0"
/>
`;
